File: src/Json/Table/IOType.fs
namespace ARCtrl.Json

open Thoth.Json.Core

open ARCtrl

module IOType =
  let encoder (io:IOType) = Encode.string <| io.ToString()

  let decoder : Decoder<IOType> = Decode.string |> Decode.andThen (
      fun s -> IOType.ofString s |> Decode.succeed
    )

File: js/Json/Table/IOType.js
import { toString } from "../../fable_modules/fable-library-js.4.22.0/Types.js";
import { string, succeed, andThen } from "../../fable_modules/Thoth.Json.Core.0.4.0/Decode.fs.js";
import { IOType } from "../../Core/Table/CompositeHeader.js";

export function encoder(io) {
    const value = toString(io);
    return {
        Encode(helpers) {
            return helpers.encodeString(value);
        },
    };
}

export const decoder = andThen((s) => succeed(IOType.ofString(s)), string);

